### PR TITLE
Integrate query proposals

### PR DESCRIPTION
**Pull request by CarlosLopezDeLara**
# Changelog

```yaml
- description: |
    Integrate queryProposals (GetProposals)
# uncomment types applicable to the change:
  type:
   - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - refactoring    # QoL changes
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

This query is relatively new in consensus and will be useful for https://github.com/IntersectMBO/cardano-cli/issues/954

# How to trust this PR

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [x] Self-reviewed the diff

### CODE REVIEW

**Review by smelc — APPROVED** [review]

File: cardano-api/internal/Cardano/Api/Query.hs [hunk 1]
```diff
@@ -296,6 +297,9 @@ data QueryInShelleyBasedEra era result where
   QueryStakeVoteDelegatees
     :: Set StakeCredential
     -> QueryInShelleyBasedEra era (Map StakeCredential (Ledger.DRep StandardCrypto))
+  QueryProposals
+    :: Set (L.GovActionId StandardCrypto)
+    -> QueryInShelleyBasedEra era (Seq (L.GovActionState (ShelleyLedgerEra era {-StandardCrypto-})))
```
Comment (carbolymer) [inline cardano-api/internal/Cardano/Api/Query.hs:302]: accidental leftover comment?
```suggestion
    -> QueryInShelleyBasedEra era (Seq (L.GovActionState (ShelleyLedgerEra era)))
```

File: cardano-api/internal/Cardano/Api/Query/Expr.hs [hunk 2]
```diff
@@ -490,3 +492,23 @@ queryAccountState cOnwards =
   queryExpr $
     QueryInEra . QueryInShelleyBasedEra (inject cOnwards :: ShelleyBasedEra era) $
       QueryAccountState
+
+queryProposals
```
Comment (carbolymer) [inline cardano-api/internal/Cardano/Api/Query/Expr.hs:496]: What happens if you pass an empty set here? Do you get all proposals or none? I can't remember what's the behaviour here. 😄 
Can you add a comment describing the behaviour for an empty set?
(we should do that for the other queries too, but that's out of scope of this PR).

**Review by carbolymer — APPROVED** [review]
LGTM! Can you test the query on the live node prior to merging?

File: cardano-api/internal/Cardano/Api/Query/Expr.hs [hunk 3]
```diff
@@ -490,3 +492,25 @@ queryAccountState cOnwards =
...
+      point
+      QueryInMode
+      r
+      IO
+      ( Either
+          UnsupportedNtcVersionError
+          (Either EraMismatch (Seq (L.GovActionState (ShelleyLedgerEra era))))
+      )
+queryProposals cOnwards govActionIds = do
+  let sbe :: ShelleyBasedEra era = inject cOnwards
```
Comment (Jimbo4350) [inline cardano-api/internal/Cardano/Api/Query/Expr.hs:513]: We have https://github.com/IntersectMBO/cardano-api/pull/690 which will allow us to remove this type annotation and the existential quantification!

**Review by Jimbo4350 — APPROVED** [review]
LGTM!

**Comment by smelc** [comment]
@carbolymer> I think it'd be nicer for the developer experience to merge this before having tested it on `cardano-testnet`, because testing requires building both `cardano-node` and `cardano-cli` locally with SRPs, which is cumbersome. I don't think there's much harm to be done if we merge this. It's not like 1. it can be very wrong and 2. users are going to jump on this feature.

**Comment by smelc** [comment]
Agreed during the team meeting to merge after the release (the one being worked on by @neilmayhew).

**Comment by smelc** [comment]
Putting to draft to make sure we don't merge until we're sure the behavior we observe in https://github.com/IntersectMBO/cardano-node/pull/6045 is the intended one (as [agreed on Slack](https://input-output-rnd.slack.com/archives/C031GFLC3DF/p1732895386605449?thread_ts=1732869358.126819&cid=C031GFLC3DF), we don't want to merge until we're sure our code (and the ledger's code) is correct).

**Comment by smelc** [comment]
We have confirmed that the behavior of the query we observe on `cardano-testnet` is correct, so merging. cc @CarlosLopezDeLara

**Pull request merged**